### PR TITLE
feat(web): Better scaling of albums and favorites

### PR DESCRIPTION
**Pull request by matthinc**
Rework albums and favorites to match different screen sizes better without scaling thumbnails too large.

Screenshot 2023-03-12 at 13 30 53

Screenshot 2023-03-12 at 13 31 03

### CODE REVIEW

File: web/src/lib/components/album-page/album-viewer.svelte [hunk 1]
```diff
@@ -479,7 +479,7 @@
 		</ControlAppBar>
 	{/if}
 
-	<section class="flex flex-col my-[160px] px-6 sm:px-12 md:px-24 lg:px-40">
+	<section class="flex flex-col my-[160px] px-6 xl:px-40">
```
Comment (matthinc) [inline web/src/lib/components/album-page/album-viewer.svelte:482]: Avoid too much "jumping" when scaling the browser window

File: web/src/lib/components/shared-components/immich-thumbnail.svelte [hunk 2]
```diff
@@ -149,8 +154,6 @@
 
 <IntersectionObserver once={false} let:intersecting>
 	<div
-		style:width={`${thumbnailSize}px`}
-		style:height={`${thumbnailSize}px`}
```
Comment (matthinc) [inline web/src/lib/components/shared-components/immich-thumbnail.svelte:153]: Is there a reason we had that? AFAIK this information was already provided by the CSS class?

File: web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte [hunk 3]
```diff
@@ -32,6 +33,12 @@
 		}
 	}
 
+	const gridLayout = `${
+		dynamicResponsiveGrid
+			? 'grid grid-cols-3 lg:grid-cols-[repeat(auto-fill,minmax(15rem,1fr))]'
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:38]: Not necessarily wrong, but why a minimum of 3 columns?

File: web/src/lib/components/shared-components/immich-thumbnail.svelte (hunk 2 above)
Comment (michelheusschen) [reply web/src/lib/components/shared-components/immich-thumbnail.svelte:153]: This probably doesn't work. Class names shouldn't contain dynamic values, use style instead

File: web/src/lib/components/shared-components/immich-thumbnail.svelte [hunk 4]
```diff
@@ -250,11 +253,9 @@
 		{#if intersecting}
 			<img
 				id={asset.id}
-				style:width={`${thumbnailSize}px`}
-				style:height={`${thumbnailSize}px`}
 				src={`/api/asset/thumbnail/${asset.id}?format=${format}&key=${publicSharedKey}`}
 				alt={asset.id}
-				class={`object-cover ${getSize()} transition-all z-0 ${getThumbnailBorderStyle()}`}
+				class={`object-cover ${getSize()} h-full v-full transition-all z-0 ${getThumbnailBorderStyle()}`}
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/immich-thumbnail.svelte:258]: `v-full` doesn't exist right?

**Review by michelheusschen — CHANGES_REQUESTED** [review]
Overall the `ImmichThumbnail` component isn't designed that well. Changing something will likely have unintended side effects in other components or pages that use it. Not saying you should refactor the whole component, but everyting should at least keep working.

Some issues I've seen are:
- Live photos and videos have wrong dimensions when starting playback by hovering over them
- The /search and /explore pages have incorrect height/width for thumbnails

File: web/src/lib/components/shared-components/immich-thumbnail.svelte (hunk 4 above)
Comment (matthinc) [reply web/src/lib/components/shared-components/immich-thumbnail.svelte:258]:
```js
		if (thumbnailSize) {
			return `w-[${thumbnailSize}px] h-[${thumbnailSize}px]`;
		}
```

This was already in `getSize`

File: web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte (hunk 3 above)
Comment (matthinc) [reply web/src/lib/components/shared-components/gallery-viewer/gallery-viewer.svelte:38]: Just IMO looks better on mobile screens.

**Pull request closed without merging**